### DISCUSSION TITLE
How to remove many colors or all unused colors at once?

### DISCUSSION THREAD

**Discussion by mikedufty** [OpenOrienteering/mapper#2073] [Questions & Answers]
Hi,
I've been given a map to modify which for some reason has hundreds of duplicate colors, red or magenta, most of which are unused.
This makes it very difficult to work with the color window as many pages of scrolling are required to move a colour to the top.
Does any one know a way to delete multiple colors at once?
I tried importing the symbols from a copy of the map, and ticking the delete unused colors option, and that did not seem to do anything.

Perhaps copy and pasting all the contents of my map into a new map might help.  Or can I go into the project file with a text editor and delete a few hundred magentas?

**Comment by dg0yt** [comment]
> I tried importing the symbols from a copy of the map, and ticking the delete unused colors option, and that did not seem to do anything.

I would take this as an indication that the colors are actually used by some symbols. Did you try to remove unused symbols first? (i.e. select all unused symbols via the context menu, then delete them.)

**Comment by mikedufty** [comment]
I didn't try that, but there are only about 30 symbols and 3000 colors.
I had a go at copying all objects into a new map, that worked to get rid of the colors, but I think I somehow introduced a slight rotation and couldn't match the location correctly.
I think I have got it working now.
I opened the .OOM file in notepad++ and deleted the 3000 lines with the spurious colors, which are conveniently right at the start of the file.   When I opened the edited file I got some error messages and got some colours in the map messed up, but re-importing the symbology from the original map fixed the display without bringing the 3000 extra colors back in.

**Reply by dl3sdo** [reply]
There is a PR for removing all unused colors: #2289
